solution-base/mongodb/how_to_upgrade.md

```diff
@@ -0,0 +1,19 @@
+# MongoDB Sharded Upgrade Process
+
+## Overview
+
+Upgrading MongoDB sharded involves several steps to ensure a smooth transition to the new version. Below is a structured guide to follow for a successful upgrade (note that all along
+I will be taking as an example the upgrade from 5 to 6):
+
+## Upgrade steps
+
+
+1. Check for all available versions of the chart using `helm search repo bitnami/mongodb-sharded --versions` to check the configuration values for a specified version: `helm show values --version 6.6.7  bitnami/mongodb-sharded`
+2. Bump mongo version in `.github/scripts/end2end/kustomization.yaml` and `solution-base/deps.yaml` files.
+3. Upgrade mongodb sharded version in charts => `CHART_MONGO_SHARDED_VERSION` in the `solution-base/mongodb/Makefile` file (e.g. `CHART_MONGO_SHARDED_VERSION:="6.6.7"`)
+4. Upgrade charts to the version targetted => `make fetch-mongodb-sharded`
+5. Manually update the patches by applying the changes manually in the new upgraded charts (note that this can be applied only to patches that target mongo sharded) =>  `git diff -- solution-base/mongodb/charts/mongodb-sharded/values.yaml > solution-base/mongodb/patches/secret-name.patch` (this operation needs to be done for every patch).
+6. Once the patches are updated, apply them to the charts with the command `make patch`.
+7. Applying the changes introduced by the upgrade can then be done (by checking the release notes for the new version e.g `https://www.mongodb.com/docs/v6.0/release-notes/6.0/`)
+8. The CI has to pass for the update to be considered effective.
+9. Perform an upgrade check with the product(s) using this Zenko version, before merging the upgrade PR. Some changes may be required at upper-levels (e.g., update mongosh commands, or add new logic in the upgrade).
```

Comment:
nothing specific here. This applies to every code change: the ci has to pass, and you need to test, considering upgrades when appropriate.

Reply:
IMHO as this is a guide , we can keep it because for mongo the upgrade is mandatory not optional to make sure everything is working fine

Reply:
the previous statement undermines this:
> The CI has to pass for the update to be considered effective.

Reply:
> nothing specific to mongo upgrades here. This applies to every code change: the ci has to pass, and you need to test, considering upgrades when appropriate.
...
> As for any change, it must first be tested: the dev must deploy his change, and indeed in case of mongo also try an upgrade (though the upgrade part is probably not so much more critical than other changes)

To me this is debatable, today we test nothing in Artesca CI that comes from Zenko, beyond the UI and its deployment. It should ideally not be required to test every change we introduce in Zenko in Artesca. And actually, we do not test everything we integrate. Also, this process may be done by someone else in the future, without ocntext of all the problems we faced till now.

For MongoDB, as per my experience, it may pass Zenko CI, Artesca CI but fail the nightly tests, or we could find issues with the new k8s version, or face upgrade issues not visible in the CI (everything passes, but in fact it is not working properly). So the main point here is really to stress that upgrading mongodb is not trivial and complete testing must be done before.

Maybe we can rephrase it better, but I strongly suggest that we keep this, and don't hide it under the "good practises", as it is not trivial...

Reply:
my point is not necessarily to remove it, but that either we need to give an "exhautive" test plan, or none at all : but saying "ci has to pass ; you must perform an upgrade" is opening the door to only perform an upgrade...

i.e. upgrading without checking that all _features_ are still working (esp. oplog management is notoriously picky, and we have quite a few other use cases: custom listing & index management in lifecycle, indexes in sorbet...)

→ if the point is about listing all the test that need to be performed, then the list must be extended significantly
→ if the point is only about the need to perform an upgrade, then it should be reworded indeed. Maybe just have a generic "_test all features to verify they are not affected by the new version; check the logs to verify there is no "silent errors", and don't forget to check that upgrade is working as well_"
→ the line about the CI is really out of place, if you really want to mention CI it should be the other way : "_Passing CI tests is not enough, you must also ...._"